fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java

```diff
@@ -303,6 +304,9 @@ public static void checkPartitionValuesExistForAddListPartition(OlapTable olapTa
 
     public static int calBucketNumAccordingToBackends() {
         int backendNum = GlobalStateMgr.getCurrentSystemInfo().getBackendIds().size();
+        if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
+            backendNum = backendNum + GlobalStateMgr.getCurrentSystemInfo().getComputeNodeIds(false).size();
```

Comment:
add `// TODO:` or `// FIXME: `, we need to revisit this code and define a proper behavior for shared data mode.